src/dfx/src/actors/pocketic.rs

```diff
@@ -323,28 +319,43 @@ fn pocketic_start_thread(
 
 #[cfg(unix)]
 #[tokio::main(flavor = "current_thread")]
-async fn initialize_pocketic(port: u16, state_dir: PathBuf, logger: Logger) -> DfxResult<usize> {
+async fn initialize_pocketic(
+    port: u16,
+    replica_config: &ReplicaConfig,
+    logger: Logger,
+) -> DfxResult<usize> {
     use pocket_ic::common::rest::{
-        CreateInstanceResponse, ExtendedSubnetConfigSet, InstanceConfig, RawTime, SubnetSpec,
+        AutoProgressConfig, CreateInstanceResponse, ExtendedSubnetConfigSet, InstanceConfig,
+        RawTime, SubnetSpec,
     };
     use reqwest::Client;
     use time::OffsetDateTime;
     let init_client = Client::new();
     debug!(logger, "Configuring PocketIC server");
+    let mut subnet_config_set = ExtendedSubnetConfigSet {
+        nns: Some(SubnetSpec::default()),
+        sns: Some(SubnetSpec::default()),
+        ii: None,
+        fiduciary: None,
+        bitcoin: None,
+        system: vec![],
+        verified_application: vec![],
+        application: vec![],
+    };
+    match replica_config.subnet_type {
+        ReplicaSubnetType::Application => subnet_config_set.application.push(<_>::default()),
+        ReplicaSubnetType::System => subnet_config_set.system.push(<_>::default()),
+        ReplicaSubnetType::VerifiedApplication => {
+            subnet_config_set.verified_application.push(<_>::default())
+        }
+    }
     let resp = init_client
         .post(format!("http://localhost:{port}/instances"))
         .json(&InstanceConfig {
-            subnet_config_set: ExtendedSubnetConfigSet {
-                nns: Some(SubnetSpec::default()),
-                sns: Some(SubnetSpec::default()),
-                ii: None,
-                fiduciary: None,
-                bitcoin: None,
-                system: vec![],
-                application: vec![SubnetSpec::default()],
-            },
-            state_dir: Some(state_dir),
+            subnet_config_set,
+            state_dir: Some(replica_config.state_manager.state_root.clone()),
             nonmainnet_features: true,
+            log_level: None,
```

Comment:
Can't we set this, too, from the dfx replica config?

Reply:
It is already set above in RUST_LOG. Should that be moved here?